app/code/local/Magmodules/Webwinkelconnect/controllers/IndexController.php

```diff
@@ -48,4 +47,41 @@ public function indexAction()
         }
     }
 
+    public function syncAction(): void {
+        $request_data = file_get_contents('php://input');
+        if (!$request_data) {
+            $this->getResponse()->setHttpResponseCode(400)->setBody('Empty request data');
+            return;
+        }
+        if (!$request_data = json_decode($request_data, true)) {
+            $this->getResponse()->setHttpResponseCode(400)->setBody('Invalid JSON data provided');
+            return;
+        }
+
+        if (!isset($request_data['webshop_id']) || !isset($request_data['api_key'])) {
+            $this->getResponse()->setHttpResponseCode(403)->setBody('Missing one or more credential fields');
+            return;
+        }
+
+        $helper = Mage::helper('webwinkelconnect');
+
+        if (!$helper->hasCorrectCredentials(strval($request_data['webshop_id']), strval($request_data['api_key']))) {
+            $this->getResponse()->setHttpResponseCode(403)->setBody('Empty or incorrect credential fields');
+            return;
+        }
+
+        if (!Mage::getModel('catalog/product')->load(strval($request_data['product_review']['product_id']))->getId()) {
+            $this->getResponse()->setHttpResponseCode(404)->setBody(sprintf('Could not find product with ID (%d)', $request_data['product_review']['product_id']));
+            return;
+        }
```

Comment:
Would move this to the `syncProductReview` implementation so you don't have to repeat the retrieval and sanitization of the product ID.